### PR TITLE
Fixed error of Tether getting destroy call second time (Tether 1.3.1)

### PR DESCRIPTION
**Pull request by vlki**
When using latest Tether 1.3.1, which should be valid based on the ^1.1.1 dependency, calling "hide" two times in a row on tooltip after "show" ends up with `tether#destroy()` being called second time and error in Tether. Expecting that `destroy` should be called just once, I updated the code to be the case.

Test included. Minified Tether 1.3.1 added to test vendors. Without the fix the test fail in Chrome 50.0.2661.75, but not in PhantomJS 2.1.1.

### CODE REVIEW

File: js/tests/unit/tooltip.js [hunk 1]
```diff
@@ -336,6 +336,19 @@ $(function () {
...
   })
 
+  QUnit.test('should not error when hiding the tooltip second time', function (assert) {
+    assert.expect(1)
+
+    $('<a href="#" rel="tooltip" title="tooltip on toggle"/>')
+      .appendTo('#qunit-fixture')
+      .bootstrapTooltip({ trigger: 'manual' })
+      .bootstrapTooltip('show')
+      .bootstrapTooltip('hide')
```
Comment (cvrebert) [inline js/tests/unit/tooltip.js]: Wait for the `shown.bs.tooltip` event before calling `hide`

File: js/tests/unit/tooltip.js [hunk 2]
```diff
@@ -14,9 +14,11 @@ $(function () {
       $.fn.bootstrapTooltip = $.fn.tooltip.noConflict()
     },
     afterEach: function () {
+      // Dispose tooltip to make sure all tethers get destroyed after each test
+      $('.tooltip').bootstrapTooltip('dispose')
```
Comment (vlki) [inline js/tests/unit/tooltip.js]: I have added dispose to the tooltip tests afterEach so it destroys all the tether instances after each test -- before those tether instances were lingering there and the newly added test was not failing always as it fails only when the `tether#destroy` is called and is the only tether in the page.

**Comment by vlki** [comment]
@cvrebert Updated. Now the test consistently fails on both PhantomJS and in Chrome when the fix is not there.

**Comment by vlki** [comment]
Tether is now [fixed](https://github.com/HubSpot/tether/commit/2d58e7baed1ddc241fc0ebbbb3ba27c1ba4be8b9) that it is possible to call destroy more times. The error is therefore not present anymore. Close this if you don't think this update is useful.

**Comment by mdo** [comment]
Closing per latest comment. Thanks!

**Pull request closed without merging**